apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/context/TracingContext.java

```diff
@@ -102,6 +102,7 @@ public void inject(ContextCarrier carrier) {
             AbstractTracingSpan firstSpan = first();
             operationId = firstSpan.getOperationId();
             operationName = firstSpan.getOperationName();
+            carrier.setEntryApplicationInstanceId(this.segment.getApplicationInstanceId());
```

Comment:
You only set the entryApplicationInstanceId by using current segment, how to propagation through ref?